bcycle-backend/Models/Entities/GroupTrip.cs

```diff
@@ -34,5 +36,22 @@ public void Update(GroupTripRequest data)
             Participants
                 .Where(p => p.Status == status)
                 .FirstOrDefault(r => r.UserId == userId);
+
+        public async Task<GroupTripResponse> AsResponse(Func<string, Task<UserInfo>> userProvider) =>
+            new GroupTripResponse
+            {
+                Id = Id,
+                Name = Name,
+                Description = Description,
+                Host = await userProvider(HostId),
+                TripCode = TripCode,
+                StartDate = StartDate,
+                EndDate = EndDate,
+                Route = Route,
+                Participants = Participants
+                    .Select(p => userProvider(p.UserId))
```

Comment:
How do I access `ParticipantStatus` now? The UI must know whether a participant is approved or not to display the appropriate controls or to hide them.

Reply:
Right, my bad